src/librustc_mir/borrow_check/error_reporting.rs

```diff
@@ -182,13 +182,23 @@ impl<'cx, 'gcx, 'tcx> MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
             );
 
             if !is_loop_move {
-                err.span_label(
-                    span,
-                    format!(
-                        "value {} here after move",
-                        desired_action.as_verb_in_past_tense()
-                    ),
-                );
+                if used_place.is_prefix_of(&moved_place) {
+                    err.span_label(
+                        span,
+                        format!(
+                            "value {} here after partial move",
```

Comment:
Would you be able to change the message based on an `if` rather than the entire `span_label` call? ie.

```rust
err.span_label(
    span,
    format!(
        "value {} here{}",
        desired_action.as_verb_in_past_tense(),
        if is_partial_move { " after partial move" } else { "" },
    )
);
```